server.js

```diff
@@ -284,6 +284,43 @@ async function generateNicknameTag(db, nickname) {
 
    io.on('connection', socket => {
       console.log('socket connected')
+      socket._storage = {};
+      // give the user 10 seconds to authenticate
+      socket._storage.timeout = setTimeout(() => {
+         console.log('socket disconnected due to inactivity');
+         socket.disconnect(true);
+      }, 1000 * 10);
+
+      socket.on('disconnect', () => {
+         console.log('socket disconnected');
+      });
+      
+      socket.on('authenticate', async data => {
+         if (!data || !data.selector || !data.token) {
```

Comment:
Could be simplified to !data?.selector || !data?.token, coudn't be, huh?

Reply:
yes